integration/gce.sh

```diff
@@ -87,72 +87,72 @@ service docker restart
 EOF
 	# It seems we need a short delay for docker to start up, so I put this in
 	# a separate ssh connection.  This installs nsenter.
-	ssh -t $name sudo docker run --rm -v /usr/local/bin:/target jpetazzo/nsenter
+	ssh -t "$name" sudo docker run --rm -v /usr/local/bin:/target jpetazzo/nsenter
 }
 
 function copy_hosts {
 	hostname=$1
 	hosts=$2
-	cat $hosts | ssh -t "$hostname" "sudo -- sh -c \"cat >>/etc/hosts\""
+	ssh -t "$hostname" "sudo -- sh -c \"cat >>/etc/hosts\"" < "$hosts"
 }
 
 # Create new set of VMs
 function setup {
 	destroy
 
-	names="$(vm_names)"
-	gcloud compute instances create $names --image $TEMPLATE_NAME --zone $ZONE
-	gcloud compute config-ssh --ssh-key-file $SSH_KEY_FILE
+	names=( $(vm_names) )
```

Comment:
Because it's the only way to be safe.

```bash
#!/bin/bash

foo=( "foo" "bar baz" "qux" )

foo_str="${foo[@]}"

python -c "import sys; print sys.argv[1:]" $foo
# Output: ['foo']

python -c "import sys; print sys.argv[1:]" "${foo[@]}"
# Output: ['foo', 'bar baz', 'qux']

python -c "import sys; print sys.argv[1:]" ${foo[*]}
# Output: ['foo', 'bar', 'baz', 'qux']

python -c "import sys; print sys.argv[1:]" $foo_str
# Output: ['foo', 'bar', 'baz', 'qux']
```